packages/react-html/src/__tests__/ReactHTMLServer-test.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+global.TextDecoder = require('util').TextDecoder;
+global.TextEncoder = require('util').TextEncoder;
+
+let React;
+let ReactHTML;
+
+describe('ReactHTML', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    // We run in the react-server condition.
+    jest.mock('react', () => require('react/react.react-server'));
+    jest.mock('react-html', () =>
+      require('react-html/react-html.react-server'),
+    );
+
+    React = require('react');
+    ReactHTML = require('react-html');
+  });
+
+  it('should be able to render a simple component', async () => {
+    function Component() {
+      // We can't use JSX because that's client-JSX in our tests.
+      return React.createElement('div', null, 'hello world');
+    }
+
+    const html = await ReactHTML.renderToMarkup(React.createElement(Component));
+    expect(html).toBe('<div>hello world</div>');
+  });
+
+  it('supports the useId Hook', async () => {
```

Comment:
❤️

Reply:
The main purpose of this test is to ensure the shared internals are wired up properly but it's also really the only Hook that does something.